src/main/java/com/odysseusinc/arachne/datanode/controller/admin/BaseAdminController.java

```diff
@@ -133,4 +158,109 @@ public JsonResult checkAtlasConnection(@PathVariable("id") Long id) {
         return result;
     }
 
+    @ApiOperation(value = "Invalidate all unfinished analyses")
+    @RequestMapping(value = Constants.Api.Analysis.INVALIDATE_ALL_UNFINISHED, method = RequestMethod.POST)
+    public Integer invalidateAllUnfinishedAnalyses(final Principal principal) throws PermissionDeniedException {
+
+        if (principal == null) {
+            throw new AuthException("user not found");
+        }
+
+        return analysisService.invalidateAllUnfinishedAnalyses(getUser(principal));
+    }
+
+    @ApiOperation(value = "list submissions")
```

Comment:
Shouldn't this method be in AnalysisController?

Reply:
The same as above